be1-go/channel/lao/mod_test.go

```diff
@@ -67,22 +180,139 @@ func TestBaseChannel_RollCallOrder(t *testing.T) {
 	}
 }
 
-func TestBaseChannel_ConsensusIsCreated(t *testing.T) {
-	// Create the hub
+func TestLAOChannel_Publish_LaoUpdate(t *testing.T) {
 	keypair := generateKeyPair(t)
+	publicKey64 := base64.URLEncoding.EncodeToString(keypair.publicBuf)
 
 	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
 	require.NoError(t, err)
 
-	// Create the messages
-	numMessages := 1
+	m := message.Message{MessageID: "0"}
+	channel := NewChannel("fzJSZjKf-2cbXH7kds9H8NORuuFIRLkevJlN7qQemjo=", fakeHub, m, nolog)
 
-	messages := make([]message.Message, numMessages)
+	// Create an update lao message
+	relativePath := filepath.Join("..", "..", "..", "protocol",
+		"examples", "messageData", "lao_update")
 
-	messages[0] = message.Message{MessageID: "0"}
+	file := filepath.Join(relativePath, "lao_update.json")
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	bufb64 := base64.URLEncoding.EncodeToString(buf)
+
+	m1 := message.Message{
+		Data:              bufb64,
+		Sender:            publicKey64,
+		Signature:         "h",
+		MessageID:         messagedata.Hash(bufb64, publicKey64),
+		WitnessSignatures: []message.WitnessSignature{},
+	}
+
+	relativePathPub := filepath.Join("..", "..", "..", "protocol",
+		"examples", "query", "publish")
+
+	filePublish := filepath.Join(relativePathPub, "publish.json")
+	bufPub, err := os.ReadFile(filePublish)
+	require.NoError(t, err)
+
+	var messagePublish method.Publish
+
+	err = json.Unmarshal(bufPub, &messagePublish)
+	require.NoError(t, err)
+
+	messagePublish.Params.Message = m1
+
+	require.NoError(t, channel.Publish(messagePublish))
```

Comment:
The test a rather weak, as we only check for the absence of an error, and not expected actions performed. 

We can't do better right now and we don't have to change it for this PR, but this is an interesting exercice: could you think about how to change the channel implementation, so that it can be better tested ? In this case we want to check that, provided with a publish/lao#update message, the channel calls the expected function associated to the message data.